include/nbl/builtin/hlsl/functional.hlsl

```diff
@@ -1,21 +1,33 @@
-// Copyright (C) 2022 - DevSH Graphics Programming Sp. z O.O.
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
 // This file is part of the "Nabla Engine".
 // For conditions of distribution and use, see copyright notice in nabla.h
-#ifndef _NBL_BUILTIN_HLSL_BINOPS_INCLUDED_
-#define _NBL_BUILTIN_HLSL_BINOPS_INCLUDED_
+#ifndef _NBL_BUILTIN_HLSL_FUNCTIONAL_INCLUDED_
+#define _NBL_BUILTIN_HLSL_FUNCTIONAL_INCLUDED_
 
 namespace nbl
 {
 namespace hlsl
 {
-namespace binops
-{
+#ifndef __HLSL_VERSION // CPP
+
+template<class T> using bit_and = std::bit_and;
+template<class T> using bit_or = std::bit_or;
+template<class T> using bit_xor = std::bit_xor;
+template<class T> using plus = std::plus;
+template<class T> using multiplies = std::multiplies;
+template<class T> using greater = std::greater;
+template<class T> using less = std::less;
+template<class T> using greater_equal = std::greater_equal;
+template<class T> using less_equal = std::less_equal;
```

Comment:
the C++ `std` versions do not have `identity` method, so I'd inherit from them instead of `using` alias